ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java

```diff
@@ -121,16 +119,7 @@ private void initTableRuntime(TableRuntimeMeta tableRuntimeMeta) {
           || tableRuntime.getOptimizingStatus() == OptimizingStatus.PENDING) {
         schedulingPolicy.addTable(tableRuntime);
       } else if (tableRuntime.getOptimizingStatus() != OptimizingStatus.COMMITTING) {
-        TableOptimizingProcess process = new TableOptimizingProcess(tableRuntimeMeta);
-        process.getTaskMap().entrySet().stream()
-            .filter(
-                entry ->
-                    entry.getValue().getStatus() == TaskRuntime.Status.SCHEDULED
-                        || entry.getValue().getStatus() == TaskRuntime.Status.ACKED)
-            .forEach(entry -> executingTaskMap.put(entry.getKey(), entry.getValue()));
-        process.getTaskMap().values().stream()
-            .filter(task -> task.getStatus() == TaskRuntime.Status.PLANNED)
-            .forEach(taskQueue::offer);
+        tableQueue.offer(new TableOptimizingProcess(tableRuntimeMeta));
```

Comment:
When starting OptimizingQueue, we should consider the status of processing some tasks, such as failures. If there are still some tasks failing and the process has not finished yet, the table will block in the optimizing state. This is the reason why I put the failed tasks into the retry queue in PR #2280 . Please also note the status of other tasks so that they can be handled correctly.

Reply:
Yeah, that's correct for earlier codes.

For this PR, the situations of failure task status in running process have been eliminated. What you encountered in #2280 is task failure and task reset or process close operation  are not in one transaction which should be. this is fixed by calling doAsTransaction in TaskRuntime.complete...  check this out